### PR TITLE
Call default on TCA

### PR DESCRIPTION
**Pull request by MiksuJak**

### CODE REVIEW

File: test/truefi2/lines-of-credit/TrueCreditAgency.test.ts [hunk 1]
```diff
@@ -906,6 +906,53 @@ describe('TrueCreditAgency', () => {
     })
   })
 
+  describe.only('enterDefault', () => {
```
Comment (skalermo) [inline test/truefi2/lines-of-credit/TrueCreditAgency.test.ts:909]:
```suggestion
  describe('enterDefault', () => {
```

**Review by barrutko — DISMISSED** [review]
LGTM

**Review by skalermo — APPROVED** [review]

**Review by vanruch — DISMISSED** [review]

File: test/truefi2/lines-of-credit/TrueCreditAgency.test.ts [hunk 2]
```diff
@@ -906,6 +906,53 @@ describe('TrueCreditAgency', () => {
...
+          .to.be.revertedWith('TrueCreditAgency: Borrower status has to be ineligible to default')
+      })
+    })
+
+    describe('makes LoC repaid from TCA point of view', () => {
+      beforeEach(async () => {
+        await timeTravel(MONTH + DAY * 3 + 1)
+      })
+
+      it('reduces principal debt', async () => {
```
Comment (vanruch) [inline test/truefi2/lines-of-credit/TrueCreditAgency.test.ts:942]:
```suggestion
      it('reduces principal debt to 0', async () => {
```

File: test/truefi2/lines-of-credit/TrueCreditAgency.test.ts [hunk 3]
```diff
@@ -906,6 +906,53 @@ describe('TrueCreditAgency', () => {
...
+        expect(await creditAgency.borrowed(tusdPool.address, borrower.address)).to.eq(1000)
+        await creditAgency.enterDefault(tusdPool.address, borrower.address)
+        expect(await creditAgency.borrowed(tusdPool.address, borrower.address)).to.eq(0)
+      })
+
+      it('reduces interest', async () => {
```
Comment (vanruch) [inline test/truefi2/lines-of-credit/TrueCreditAgency.test.ts:948]:
```suggestion
      it('reduces interest to 0', async () => {
```

**Review by barrutko — APPROVED** [review]

**Review by skalermo — APPROVED** [review]

File: contracts/truefi2/TrueCreditAgency.sol [hunk 4]
```diff
@@ -391,6 +391,27 @@ contract TrueCreditAgency is UpgradeableClaimable, ITrueCreditAgency {
...
+    function enterDefault(ITrueFiPool2 pool, address borrower) external {
+        require(borrowed[pool][borrower] > 0, "TrueCreditAgency: Borrower does not have any debt in pool");
+        require(
+            block.timestamp >= nextInterestRepayTime[pool][borrower].add(creditOracle.gracePeriod()),
+            "TrueCreditAgency: Borrower can still repay the debt"
+        );
+        require(
+            creditOracle.status(borrower) == ITrueFiCreditOracle.Status.Ineligible,
+            "TrueCreditAgency: Borrower status has to be ineligible to default"
+        );
```
[the remark this answers is not in the document]
Comment (yuchenlintt) [reply contracts/truefi2/TrueCreditAgency.sol:406]: I'm pretty certain the answer to this question is already planned in a follow-up PR, but I want to make sure it stays on top of our minds.

What prevents the borrower from withdrawing again after default? We need the borrowing mutex to unlock() + lock(address(debtToken)) in a follow-up PR.

**Review by yuchenlintt — COMMENTED** [review]
This looks like a great start!

However there are some pretty major deviations from our design that we should resolve in follow-up PRs.

I think you made the right call to merge this PR quickly though, since we want to stabilize an interface for SAFU + liquidator + DebtToken to interact with. I'll file tasks to address my comments.

**Pull request merged**